hazelcast/src/main/java/com/hazelcast/internal/config/override/ExternalConfigurationOverride.java

```diff
@@ -83,7 +83,7 @@ public ClientConfig overwriteClientConfig(ClientConfig config) {
                   properties.entrySet().stream()
                     .filter(e -> !unprocessed.containsKey(e.getKey()))
                     .map(e -> {
-                        if (e.getKey().equals("hazelcast.licensekey")) {
+                        if (e.getKey().equals("hazelcast.licensekey") && !e.getValue().isEmpty()) {
```

Comment:
Can we make this more resilient to invalid input? The code below will still throw `StringIndexOutOfBoundsException` if input license key is `asdf`. It seems `originalKeyPart` needs to have length at least 5 for the `substring`s to work.

Reply:
Would be great to provide unit tests for empty/null/short license keys